Tests/LeafErrorMiddlewareTests/CustomMappingDefaultGeneratorTests.swift

```diff
@@ -0,0 +1,152 @@
+import LeafErrorMiddleware
+@testable import Logging
+import Vapor
+import XCTest
+
+class CustomMappingDefaultGeneratorTests: XCTestCase {
+    // MARK: - Properties
+
+    var app: Application!
+    var viewRenderer: ThrowingViewRenderer!
+    var logger = CapturingLogger()
+    var eventLoopGroup: EventLoopGroup!
+
+    // MARK: - Overrides
+
+    override func setUpWithError() throws {
+        eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        viewRenderer = ThrowingViewRenderer(eventLoop: eventLoopGroup.next())
+        LoggingSystem.bootstrapInternal { _ in
+            self.logger
+        }
+        app = Application(.testing, .shared(eventLoopGroup))
+
+        app.views.use { _ in
+            self.viewRenderer
+        }
+
+        func routes(_ router: RoutesBuilder) throws {
+            
+            router.get("404") { _ -> HTTPStatus in
+                throw Abort(.notFound)
+            }
+
+            router.get("403") { _ -> Response in
+                throw Abort(.forbidden)
+            }
+            
+            router.get("303") { _ -> Response in
+                throw Abort.redirect(to: "ok")
+            }
+
+            router.get("serverError") { _ -> Response in
+                throw Abort(.internalServerError)
+            }
+
+            router.get("unknownError") { _ -> Response in
+                throw TestError()
+            }
+
+            router.get("unauthorized") { _ -> Response in
+                throw Abort(.unauthorized)
+            }
+
+            router.get("401withReason") { _ -> HTTPStatus in
+                throw Abort(.unauthorized, reason: "You need to log in")
+            }
+
+        }
+
+        try routes(app)
+
+        app.middleware.use(LeafErrorMiddlewareDefaultGenerator.build(errorMappings: [
+            .notFound: "404",
+            .unauthorized: "401",
+            .forbidden: "403",
+            // Verify that non-error mappings are ignored
+            .seeOther: "303"
```

Comment:
We already do this in the other tests, it seems that we should be using the default generator here?

Reply:
This test suite is focused on using the Default Generator accepting the custom mapping, whereas other test suites either use the custom context generator or the default generator sans custom mappings